pymedphys/_dicom/anonymisation_strategy.py

```diff
@@ -0,0 +1,151 @@
+
+
+def _get_LO_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("LO", current_value)
+
+
+def _get_LT_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("LT", current_value)
+
+
+def _get_OB_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("OB", current_value)
+
+
+def _get_OB_or_OW_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("OB or OW", current_value)
+
+
+def _get_OW_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("OW", current_value)
+
+
+def _get_PN_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("PN", current_value)
+
+
+def _get_SH_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("SH", current_value)
+
+
+def _get_SQ_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("SQ", current_value)
+
+
+def _get_ST_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("ST", current_value)
+
+
+def _get_TM_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("TM", current_value)
+
+
+def _get_UI_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("UI", current_value)
+
+
+def _get_US_anonymous_hardcode_replacement_value(current_value):
+    return _get_vr_anonymous_hardcode_replacement_value("US", current_value)
+
+
+ANONYMISATION_HARDCODE_DISPATCH = dict(
+    {
+        "AE": _get_AE_anonymous_hardcode_replacement_value,
+        "AS": _get_AS_anonymous_hardcode_replacement_value,
+        "CS": _get_CS_anonymous_hardcode_replacement_value,
+        "DA": _get_DA_anonymous_hardcode_replacement_value,
+        "DS": _get_DS_anonymous_hardcode_replacement_value,
+        "DT": _get_DT_anonymous_hardcode_replacement_value,
+        "LO": _get_LO_anonymous_hardcode_replacement_value,
+        "LT": _get_LT_anonymous_hardcode_replacement_value,
+        "OB": _get_OB_anonymous_hardcode_replacement_value,
+        "OB or OW": _get_OB_or_OW_anonymous_hardcode_replacement_value,
+        "OW": _get_OW_anonymous_hardcode_replacement_value,
+        "PN": _get_PN_anonymous_hardcode_replacement_value,
+        "SH": _get_SH_anonymous_hardcode_replacement_value,
+        "SQ": _get_SQ_anonymous_hardcode_replacement_value,
+        "ST": _get_ST_anonymous_hardcode_replacement_value,
+        "TM": _get_TM_anonymous_hardcode_replacement_value,
+        "UI": _get_UI_anonymous_hardcode_replacement_value,
+        "US": _get_US_anonymous_hardcode_replacement_value,
+    }
+)
```

Comment:
To remove the duplicated logic that is found in lines 58-127 can we change lines 58-151 to something like the following:

```python
KEYS_TO_OVERRIDE = [
    "AE", "AS", "CS", "DA", "DS", "DT",
    "LO", "LT", "OB", "OB or OW", "OW",
    "PN", "SH", "SQ", "ST", "TM", "UI", 
    "US"
]

ANONYMISATION_HARDCODE_DISPATCH = {
    key: functools.partial(_get_vr_anonymous_hardcode_replacement_value, value_representation=key)
    for key in KEYS_TO_OVERRIDE 
}
```

Reply:
yes.  but this approach will not likely be applicable to more complex strategies (e.g. pseudonymisation), or at least not fully applicable, and not without it's own twisting around to avoid vr specific logic.
it is applicable here.  I wasn't aware of partial functions in python before.  Learn something new every day :-)

Reply:
> or at least not fully applicable, and not without it's own twisting around to avoid vr specific logic.

Yeah, certainly don't use it if it will require twisting.

> I wasn't aware of partial functions in python before. Learn something new every day :-)

Yeah, it certainly has a well defined place. It can be overused (like anything), but it can result in neat solutions. I am always amazed how much I learn through Open Source :).